extensions/warp-ipfs/src/store/conversation.rs

```diff
@@ -128,11 +127,18 @@ impl ConversationDocument {
     }
 
     pub fn conversation_type(&self) -> ConversationType {
-        match self.settings {
-            ConversationSettings::Direct(_) => ConversationType::Direct,
-            ConversationSettings::Group(_) => ConversationType::Group,
+        if self.recipients.len() <= 2 {
+            ConversationType::Direct
+        } else {
+            ConversationType::Group
```

Comment:
I added `ConversationType::from_recipients` method, and call it from `conversation_type` methods on both `ConversationDocument` and `Conversation` to avoid this code duplication.

Reply:
Maybe this is what you meant we should do all along: I added conversation_type to Conversation and ConversationDocument. This way it can be set on creation. Most of the remaining broken tests were due to incorrect determination of conversation_type. I incorrectly assumed this could be inferred from the number of participants, but there are test cases that create group chats with only 1 or 2 participants.

Reply:
Needless to say, I removed `ConversationType::from_recipients` and `conversation_type()` methods now return the value of the field on their respective structs.